main/src/ca/uwaterloo/flix/language/phase/jvm/GenMainClass.scala

```diff
@@ -49,7 +49,8 @@ object GenMainClass {
     * Throws `InternalCompilerException` if the type  of `defn` is not `Unit -> Unit`.
     */
   private def checkMainType(defn: Def): Unit = (defn.cparams, defn.fparams, defn.tpe) match {
-    case (Nil, List(FormalParam(_, _, MonoType.Unit, _)), MonoType.Unit) => ()
+    // todo this is too broad
+    case (Nil, List(FormalParam(_, _, MonoType.Unit, _)), MonoType.Object) => ()
```

Comment:
This check shouldnt be needed, the main function is checked in the frontend

Reply:
Could it be removed?

Reply:
it could yea